staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_aggregator.go

```diff
@@ -47,36 +48,65 @@ type openAPIAggregator struct {
 	// provided for dynamic OpenAPI spec
 	openAPIService *handler.OpenAPIService
 
-	// Aggregator's OpenAPI spec (holds apiregistration group).
-	aggregatorOpenAPISpec *spec.Swagger
+	contextMapper request.RequestContextMapper
+}
 
-	// Local (in process) delegate's OpenAPI spec.
-	inProcessDelegatesOpenAPISpec *spec.Swagger
+func (s *openAPIAggregator) LoadAndAddLocalSpec(localHandler http.Handler, name string, filterPaths []string) error {
+	spec, err := s.downloadOpenAPISpec(localHandler)
+	if err != nil {
+		return err
+	}
+	if filterPaths == nil {
```

Comment:
do we actually need this method? It has side-effects and I wonder whether we should concentrate that to the logic below with the actual loop.